scalameta/trees/shared/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala

```diff
@@ -363,21 +363,15 @@ object TreeSyntax {
           case _ => m(Path, s(p(SimpleExpr, t.qual), if (guessIsPostfix(t)) " " else ".", t.name))
         }
       case t: Term.Interpolate =>
-        def needBraces(id: String, charAfter: Option[Char]): Boolean = {
-          val startsWithUnderscore = id.startsWith("_")
-          startsWithUnderscore || {
-            val isOpId = isOperatorPart(id.head)
-            val charAfterIsOp = charAfter.exists(isOperatorPart)
-            val charAfterIsIdPart =
-              charAfter.exists(chr => isNameStart(chr) || Character.isDigit(chr))
-            if (isOpId) charAfterIsOp else charAfterIsIdPart
-          }
+        def needBraces(id: String, nextPart: String): Boolean = {
+          val idHead = id.head
+          id == '_' || !Character.isUnicodeIdentifierStart(idHead) ||
```

Comment:
```suggestion
          idHead == '_' || !Character.isUnicodeIdentifierStart(idHead) ||
```
otherwise it's always false, no?

Reply:
good eye! apparently, `_` is not a valid unicode identifier start:
<img width="625" alt="Screenshot 2024-01-05 at 09 21 28" src="https://github.com/scalameta/scalameta/assets/7529386/76657d09-5567-4e97-bab2-ec64b592d755">

but that presents an opportunity to simplify further. i have also found another peculiarity, so minor changes in the next push.

Reply:
done, please take a look